webpush/encrypt.go

```diff
@@ -250,15 +262,14 @@ func newInfo(infoType string, context []byte) []byte {
 //
 // This is a partial implementation of HKDF tailored to our specific purposes.
 // In particular, for us the value of N will always be 1, and thus T always
-// equals HMAC-Hash(PRK, info | 0x01).
+// equals HMAC-Hash(PRK, info | 0x01). This is true because the maximum output
+// length we need/allow is 32.
 //
 // See https://www.rfc-editor.org/rfc/rfc5869.txt
-func hkdf(salt, ikm, info []byte, length int) []byte {
+func hkdf(salt, ikm, info []byte, length int) ([]byte, error) {
 	// HMAC length for SHA256 is 32 bytes, so that is the maximum result length.
-	// HKDF defines a way to have longer lengths, but our partial implementation
-	// will just truncate the result.
 	if length > 32 {
-		length = 32
+		return nil, fmt.Errorf("Can only produce HKDF outputs up to 32 bytes long")
```

Comment:
I think in this case it shouldn't do either. Certainly not panic, nor should it limit length since all callers are internal, you know exactly what you're doing. There's no point in setting foolproof guards against yourself, there are tests for this :)

If an internal call to `hkdf()` depends on some user input, then the caller of hkdf should verify that user input and return an error straight away if needed, without even calling hkdf.